client/remote_access.go

```diff
@@ -115,8 +115,26 @@ func (target RemoteAccessTarget) String() string {
 	return base
 }
 
-// parseNetworkPort parses a network port string.
-func parseNetworkPort(portString string) (uint16, error) {
+// parseLocalNetworkPort parses a network port string and accepts either a port number or "stdio".
+func parseLocalNetworkPort(portString string) (string, error) {
```

Comment:
How about we keep the `parseNetworkPort` as is, and only add:

```golang

// parseLocalNetworkPort parses a network port string and accepts either a port number or "stdio".
func parseLocalNetworkPort(portString string) (string, error) {
	if portString == "stdio" {
		return portString, nil
	}

	if _, err := parseNetworkPort(portString); err != nil {
		return "", fmt.Errorf("invalid port number")
	}

	return portString, nil
}
```

That way we avoid code duplication and make sure that rules for numeric port validation are consistent for local and remote ports.

Reply:
I see that you removed the `parseLocalNetworkPort` completely, but using "stdio" as a remote port is probably not going to work. If it would work and there is a use case there, I'm fine having both ends accept "stdio", but if the system won't work then, we probably should still check to make sure that incorrect values cannot be set.